### ISSUE TITLE
PortalTargetGroup - A component to show multiple sources at the same time

### ISSUE THREAD

**Issue by LinusBorg** [LinusBorg/portal-vue#76]
(For previous discussions, see #26)

### Why do we need it?

One of PortalVue's major usecases is moving components like Modals, dropdowns, tooltips and other absolutely positioned stuff out of the deeply nested DOM structure and directly up below the `<body>` tag.

This works well if the developer can control/gurantee that only one e.g. Modal is sent to a PortalTarget at a time, but many interfaces allow for multiple such components to be active at the same time.

One can work around that manually to a degree, e.g. by setting up multiple PortalTargets in advance, but that's not very nice. So we need a solution that provides an outlet like PortalTarget, but can show content from many `<portal>` sources at the same time.

### API

My current plans should solve that, but in a different way: I'm working on a `PortalTargetGroup` component, which can render any number PortalTargets at once. It will go something like this:
```html
<portal name="edit-user-modal" group="modals">
  <!-- slot content -->
</portal>

<portal name="edit-user-picture-modal" group="modals">
  <!-- slot content -->
</portal>
```
Those two portals should be allowed to render at the same time, in a Target, and that's what `PortalTargetGroup` does:
```html
<portal-target-group  name="modals"/>
```

It receives the slot content for all targets in the group specified by its name, and render them one after the other.

That architecture allows to achieve this feature with only minor internal changes and one new component.

It should also be able to render as a `TransitionGroup` so incoming and leaving targets can be animated:

```html
<portal-target-group  
  name="modals"
  :transition="{name: fade }"
/>
```
And it will support basic sorting of the targets:
```html
<portal-target-group  
  name="modals"
  sort="name-asc" // or:
  sort="index-desc" // or:
  sort="udpated-asc" // or:
  :sort="sortFn"  // a function to sort manually.
/>
```

That's the outline, and it's going good so far. Maybe I can get it working during the coming weekend.

If tests go well, a relase in a week or two is possible.

**Comment by LinusBorg** [comment]
Update: This idea was a dead end, but fortunately, there's a PR in the works tackling this from another angle, and it's looking great: #80

**Comment by vms82** [comment]
not sure if this is the appropriate place to ask so excuse me if this is the wrong place , i have the following use case, i render different components in a container component that basicly just receive events with component name and some data and render them inside a single div with v-for, each component has collapse state which shrinks the entire thing and change its flex order so it move up letting the rest non collapsed components to be below the "collapsed" components, was wondering can i use this to port all collapsed components to separated div not nessecery outside of app so they can fit in a small div with horizontal scroll bar, (also does the component gets recreated when its ported? some of my components can be ajax heavy and i would like to avoid making unnececery db calls)

**Comment by LinusBorg** [comment]
That doesn't soujnd like a good use case for a portal, no. I suggest to ask on our forum for other ideas on how to solve this challenge: forum.vuejs.org

**Comment by LinusBorg** [comment]
closed with #80
